src/Container/ApplicationConfigInjectionDelegator.php

```diff
@@ -23,15 +24,20 @@ class ApplicationConfigInjectionDelegator
      * Decorate an Application instance by injecting routes and/or middleware
      * from configuration.
      *
-     * @return mixed|Application Typically, this should return an Application
-     *     instance. However, if the delegator is attached to some other service,
-     *     there is a possibility it will return another type.
+     * @throws Exception\InvalidServiceException if the $callback produces
+     *     something other than an `Application` instance, as the delegator cannot
+     *     proceed with its operations.
      */
-    public function __invoke(ContainerInterface $container, string $serviceName, callable $callback)
+    public function __invoke(ContainerInterface $container, string $serviceName, callable $callback) : Application
     {
         $application = $callback();
         if (! $application instanceof Application) {
-            return $application;
+            throw new Exception\InvalidServiceException(sprintf(
```

Comment:
Why not static method in exception?

Reply:
This one hasn't had one defined ever. I can add some later, but in most cases, we'll have different messages for every single factory, making it less viable.